File: Src/Litium.Accelerator.Administration.Extensions/src/AcceleratorExtensions/interfaces/searchIndexingModel.ts
export interface ISearchIndexingModel {
    groupedTemplates: ITemplateGroupModel[],
    message: string,
}

export interface ITemplateGroupModel {
    title: string;
    templates: ITemplateModel[]
}

export interface ITemplateModel {
    title: string;
    fields: IFieldModel[],
    selectedFields: IFieldModel[],
    templateId: string,
    areaType: string,
}

export interface IFieldModel {
    title: string;
    fieldId: string,
}

File: Src/Litium.Accelerator.Administration.Extensions/ViewModels/SearchIndexingModel.cs
using System;
using System.Collections.Generic;

namespace Litium.Accelerator.Administration.Extensions.ViewModel
{
    public class SearchIndexingModel
    {
        public List<TemplateGroup> GroupedTemplates { get; set; }

        public string Message { get; set; }

        public class Template
        {
            public string Title { get; set; }
            public List<FieldGroup> Fields { get; set; }
            public List<string> SelectedFields { get; set; } = new List<string>();
            public string TemplateId { get; set; }
            public Type AreaType { get; internal set; }
        }

        public class FieldGroup
        {
            public string Title { get; set; }
            public string FieldId { get; set; }
        }

        public class TemplateGroup
        {
            public string Title { get; set; }
            public List<Template> Templates { get; set; }
        }
    }
}
